indexing-hadoop/src/main/java/io/druid/indexer/JobHelper.java

```diff
@@ -454,7 +454,25 @@ public long push() throws IOException
         .withSize(size.get())
         .withBinaryVersion(SegmentUtils.getVersionFromDir(mergedBase));
 
-    if (!renameIndexFiles(outputFS, tmpPath, finalIndexZipFilePath)) {
+    int retryAttempts = 0;
+    int retryThreshold = 5;
+    int retryGranularity = 1000;
+
+    while (retryAttempts != retryThreshold) {
+      if (renameIndexFiles(outputFS, tmpPath, finalIndexZipFilePath)) {
+        break;
+      } else {
+        retryAttempts++;
+        try {
+          Thread.sleep(retryAttempts * retryGranularity);
+        }
+        catch (InterruptedException ex) {
+          Thread.currentThread().interrupt();
+        }
+      }
+    }
+
+    if (retryAttempts == retryThreshold) {
       throw new IOException(
           String.format(
```

Comment:
I looked at `renameIndexFiles()` and `FileSystem.rename()` they seem to capture all fail cases I can think of, I'm not sure what the exact reason is in this case